earthfile2llb/listener.go

```diff
@@ -962,6 +967,12 @@ func (ssf *StringSliceFlag) Set(arg string) error {
 	return nil
 }
 
+var shellNameStringRegexp = regexp.MustCompile("^[a-zA-Z_]+[a-zA-Z0-9_]*")
+
+func checkShellName(str string) string {
```

Comment:
This function would be better suited as returning an error if something is wrong with the env var passed.

Also, I would call it `checkEnvVarName`, as it's not really related to the shell.

Reply:
With "something is wrong with the env var passed" you mean if it doesn't match against the regex, right? Or you mean something else?

Reply:
Yes - exactly

Reply:
I added the return of error, but I am not sure about what I did. From [documentation](https://golang.org/pkg/regexp/#Regexp.FindString), I got that `FindString` returns an empty string if it doesn't match. That's the reasoning behind my change. Let me know your thoughts and if you want me to change something.

Reply:
would it be possible to change `checkEnvVarName(str string) (string, error)` to only return an error? right now it's returning a string that's not being modified or used by the caller.

then it'll simplify the calling code to:
```
	var err = checkEnvVarName(envKey)
	if err != nil {
		l.err = err
		return
	}
	l.envArgKey = envKey
```

Reply:
Of course! Consider it done.